### PR TITLE
feat: introduce rbsp unscaping buffer reader

### PR DESCRIPTION
**Pull request by murillo128**
completely untested, ptal

### CODE REVIEW

File: src/h264/H26xNal.h [hunk 1]
```diff
@@ -6,16 +6,138 @@
...
 #include "tools.h"
 #include "Buffer.h"
 #include "BufferReader.h"
+#include "bitstream/BitReader.h"
 
 constexpr uint32_t AnnexBStartCode = 0x01;
 
 // H.264 NAL logic that can be shared with H.265 (mostly emulation prevention, annex B stream)
 
+class RbspReader
```
Comment (eck1001) [inline src/h264/H26xNal.h:21]: It might be nice to add somewhere that Rbsp stands for `Raw Byte Sequence Payload`.

File: src/h264/H26xNal.h [hunk 2]
```diff
@@ -6,16 +6,138 @@
...
+{
+public:
+	RbspReader(const uint8_t* data, const size_t size) :
+		data(data),
+		size(size)
+	{
+		this->pos = 0;
+	}
+
+	inline bool   Assert(size_t num) const
```
Comment (eck1001) [inline src/h264/H26xNal.h:31]: What are we asserting here? 

Is this method used to skip the input NAL for a h26x buffer?

Comment (murillo128) [reply src/h264/H26xNal.h:31]: this method is used to ensure that we have enough bytes to call PeekX method safely, we probably want to replace them with exceptions eventually. 

Note that this RbspReader as the same methods as the BufferReader so we can use them as template input for the BaseBitReader

File: src/h264/H26xNal.h (hunk 1 above)
Comment (mildsunrise) [reply src/h264/H26xNal.h:21]: I suggest documenting what this class does, something like

``` c
/**
 * This reader consumes an RBSP (usually the payload of an H.26x NALU)
 * and provides the unescaped bytes of the SODB, by skipping over
 * `emulation_prevention_three_byte`s as necessary (H.264 section 7.4.1.1 step 1).
 * Other codes (00 00 [00..02]) like start codes, as well as incomplete codes,
 * are currently passed through.
 */
```

File: src/h264/H26xNal.h [hunk 3]
```diff
@@ -6,16 +6,138 @@
...
+			{
+				//Skip input
+				i++;
+				//No zeros
+				z = 0;
+			}
+			else 
+			{
+				//Acu value
+				value = value << 1 | data[i];
```
Comment (mildsunrise) [inline src/h264/H26xNal.h:117]:
```suggestion
				value = value << 8 | data[i];
```

**Review by mildsunrise — APPROVED** [review]
yayyyy, finally ^^ thanks for implementing this.
other than the typo it lgtm. i'm sad we're copying the same bunch of code in three places though

**Comment by murillo128** [comment]
thanxs for your reivews! yes, it is sad that I had to copy the same code in 3 places, couldn't find any good idea for cleaning it up, any idea on ow to improve it?

**Pull request merged**